examples/vision/semantic_image_clustering.py

```diff
@@ -0,0 +1,576 @@
+"""
+Title: Semantic Image Clustering
+Author: [Khalid Salama](https://www.linkedin.com/in/khalid-salama-24403144/)
+Date created: 2021/02/28
+Last modified: 2021/02/28
+Description: Semantic Clustering by Adopting Nearest neighbors (SCAN) algorithm.
+"""
+
+"""
+## Introduction
+
+This example demonstrates applying the [Semantic Clustering by Adopting Nearest neighbors
+(SCAN)](https://arxiv.org/abs/2005.12320) algorithm, by Wouter Van Gansbeke et. al, on the
+[CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) images dataset. The algorithm,
```

Comment:
Hi, first 2 cents:

Suggestion:

- Since the paper says:

```
The proposed method, named SCAN (Semantic Clustering by Adopting Nearest neighbors), 
leverages the advantages of both representation and end-to-end learning approaches
```

maybe you should rephrase the intro like so:

```suggestion
This example demonstrates how to apply the [Semantic Clustering by Adopting Nearest neighbors
(SCAN)](https://arxiv.org/abs/2005.12320) method (Van Gansbeke et. al, 2020) on the
[CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html) images dataset. The algorithm,
```

(I think the format for references with several authors in ML research is usually `Last Name et al., YEAR`)

LMKWYT

cc @fchollet

Reply:
I Updated the reference style, but avoided referring to this technique as classification